### PR TITLE
Drop support for CameraFrame in ImageProcessor

### LINKED ISSUE #2061: remove support for calculating Hillas parameters in CameraFrame
see older issue: #1983 

Now that everything works consistently in the TelescopeFrame for reconstruction, we should remove the redundant option to do it in the CameraFrame.   In general doing parameters in the TelescopeFrame is better as they can be directly compared between telescope optics types.

### PR DESCRIPTION
**Pull request by StFroese**
Initially I wanted to remove the option to use the CameraFrame for calculating the Hillas parameters, but I had to change a lot of stuff and ended up removing the CameraFrame completely from the ImageProcessor.

fixes #2061

### CODE REVIEW

**Comment by kosack** [comment]
Thanks this will clean up a lot of redundant code! It's clear a lot of pieces need to be touched, so that is ok.  The only consequence is that it means pointing corrections and the focal length are "baked in" to the DL1 parameters (assuming they are applied in the CameraFrame → TelescopeFrame transition), but I think that is a small price to pay for simplicity, and if the need arises we can add a way to correct them.

**Comment by kosack** [comment]
You'll need to update some of the tutorials as well, as some were written before there was a TelescopeFrame I'm afraid!

**Review by kosack — COMMENTED** [review]
Since so much had to change, it may be a good idea for somebody to run an analysis (both regular reconstruction as well as muon analysis) on a large number of events and compare the results to the old version before we merge this.  Just in case it introduces some unexpected typos.

**Comment by maxnoe** [comment]
There is a lot of code in `ctapipe/reco/hillas_reconstructor` and `ctapipe/reco/hillas_intersection` dealing with parameters in camera frame, these should also be removed.

**Comment by StFroese** [comment]
@maxnoe should this be in this PR since this one is just ImageProcessor? I would do a separate PR for the ShowerProcessor once this is merged

File: ctapipe/reco/hillas_intersection.py [hunk 1]
```diff
@@ -191,24 +186,15 @@ def _predict(self, hillas_dict, array_pointing, telescopes_pointings=None):
...
-                cog_coords = SkyCoord(
-                    fov_lon=hillas.fov_lon,
-                    fov_lat=hillas.fov_lat,
-                    frame=telescope_frame,
-                )
-                cog_coords_nom = cog_coords.transform_to(nom_frame)
+            telescope_frame = TelescopeFrame(
+                telescope_pointing=telescopes_pointings[tel_id]
+            )
+            cog_coords = SkyCoord(
```
Comment (maxnoe) [inline ctapipe/reco/hillas_intersection.py:192]: Re our discussion about SkyCoord vs. Frames: could you also check here if doing `TelescopeFrame(fov_lon, ...)` is faster?

Comment (StFroese) [reply ctapipe/reco/hillas_intersection.py]: Yes it's faster. For 200 events:
With SkyCoord: `1.06 ± 0.01`
With Frame: `0.86±0.01`

**Comment by StFroese** [comment]
I checked the dl1 parameters for this branch and master for regular and for the muon analysis. Works fine. 
Also updated the jupyter notebooks in the docs

File: ctapipe/io/hdf5eventsource.py [hunk 2]
```diff
@@ -351,18 +349,6 @@ def _parse_sb_and_ob_configs(self):
 
         return scheduling_blocks, observation_blocks
 
-    def _is_hillas_in_camera_frame(self):
```
Comment (LukasNickel) [inline ctapipe/io/hdf5eventsource.py:354]: Just to be sure:
We want to completely drop reading files created in the CameraFrame?

File: ctapipe/reco/impact.py [hunk 3]
```diff
@@ -204,8 +204,8 @@ def get_hillas_mean(self):
         tel_num = 0
 
         for hillas in self.hillas_parameters:
-            peak_x[tel_num] = hillas.x.to(u.rad).value  # Fill up array
-            peak_y[tel_num] = hillas.y.to(u.rad).value
+            peak_x[tel_num] = hillas.fov_lon.to(u.rad).value  # Fill up array
+            peak_y[tel_num] = hillas.fov_lat.to(u.rad).value
```
Comment (LukasNickel) [inline ctapipe/reco/impact.py:208]: Its not really x and y anymore, but I guess thats fine here

File: ctapipe/image/extractor.py [hunk 4]
```diff
@@ -1051,7 +1052,10 @@ def _apply_second_pass(
         # Apply correction to 1st pass charges
         charge_1stpass = charge_1stpass_uncorrected * correction[selected_gain_channel]
 
-        camera_geometry = self.subarray.tel[tel_id].camera.geometry
+        # do it in TelescopeFrame
```
Comment (LukasNickel) [inline ctapipe/image/extractor.py:1055]: That comment is not very useful

File: docs/examples/camera_display.ipynb [hunk 5]
```diff
@@ -91,9 +92,20 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 5,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
```
Comment (LukasNickel) [inline docs/examples/camera_display.ipynb:97]: We wanted to strip all notebooks before committing, right?

File: docs/changes/2141.api.rst [hunk 6]
```diff
@@ -0,0 +1,4 @@
+Drop support for ``CameraFrame`` in ``ImageProcessor``
+
+``ImageProcessor`` now only uses the ``TelescopeFrame`` to perform all image operations.
+This also drops support for calculating the ``HillasParameters`` in ``CameraFrame``.
```
Comment (LukasNickel) [inline docs/changes/2141.api.rst:4]: And reading such files!

**Review by LukasNickel — CHANGES_REQUESTED** [review]
Looks almost done to me. Some comments:
- If we do not support loading camera frame files, we need to put that in the changelog. I am fine with it, it is bound to happen at some point
- Strip notebooks
- Fix merge conflicts

File: ctapipe/image/extractor.py (hunk 4 above)
Comment (StFroese) [reply ctapipe/image/extractor.py:1055]: I guess that was just a note for myself😅

File: ctapipe/reco/impact.py (hunk 3 above)
Comment (StFroese) [reply ctapipe/reco/impact.py:208]: It would be a small change to rename it but I'm really hesitant to do changes to the ImPACT code since #2052 is still open.

Comment (LukasNickel) [reply ctapipe/reco/impact.py:208]: Yeah, that is why I meant its fine.
Lets not touch IMPACT code more than we need to

File: ctapipe/io/hdf5eventsource.py (hunk 2 above)
Comment (StFroese) [reply ctapipe/io/hdf5eventsource.py:354]: I think so. Like you said, it will happen at some point. In this case it is just the result of getting rid of `CameraHillasParametersContainer`

File: ctapipe/image/extractor.py [hunk 7]
```diff
@@ -1051,7 +1052,9 @@ def _apply_second_pass(
         # Apply correction to 1st pass charges
         charge_1stpass = charge_1stpass_uncorrected * correction[selected_gain_channel]
 
-        camera_geometry = self.subarray.tel[tel_id].camera.geometry
+        camera_geometry = self.subarray.tel[tel_id].camera.geometry.transform_to(
+            TelescopeFrame()
```
Comment (kosack) [inline ctapipe/image/extractor.py:1056]: this might need some thought - we shouldn't be transforming the geometry for each event *inside* a calibration/image-extraction algorithm like this as that adds quite a bit of overhead.   This algorithm does not care about the frame of the camera geometry or pointing corrections, so it should not require a transform. 

Since we do eventually need to transform for the ImageProcessor, it may be better to do the transform outside of these algorithms, perhaps in the CameraCalibrator (which eventually should also apply pointing correcions), so that a single TelescopeFrame is available to all algorithms that need it.  That would avoid re-doing the transform many times.

**Review by kosack — CHANGES_REQUESTED** [review]

File: ctapipe/image/extractor.py (hunk 7 above)
Comment (maxnoe) [reply ctapipe/image/extractor.py:1056]: Yes, this should be done inside the `ProcessorTool` in `setup` after writing out the subarray description but before setting up telescope components. The telescope components should just get a subarray with cameras in `CameraFrame`.

Comment (maxnoe) [reply ctapipe/image/extractor.py:1056]: Maybe add `transform_camera_geometries` to `SubarrayDescription` returning a new Subarray with transformed geometries

Comment (kosack) [reply ctapipe/image/extractor.py:1056]: Is creating a whole new SubarrayDescription each time efficient? This was an issue I think we discussed a while back (need to find where): during analysis, the TelescopeFrame needs to be updated once per event for observations, and could be simplified to only once per run for  simulations in fixed Alt/Az mode.

The TelescopeFrame requires: 
* the latest pointing position, which changes in time in ICRS tracking mode (the default for most observations)
* pre-computed pointing corrections which could include not just a shift in RA/Dec, but also a rotation (this is not currently implemented, but will need to be done eventually)

Comment (maxnoe) [reply ctapipe/image/extractor.py:1056]: Ah, sorry, forgot about pointing corrections.

But it stands: we could also do the pointing corrections in the transformation from / to TelescopeFrame to higher frames. That would actually make the most sense to me, since pointing direction is not important for the `CameraFrame` <-> `TelescopeFrame` transformation.

To me that means we should transform the whole subarray *once* to `TelescopeFrame` and apply corrections in the transformation `AltAz` <-> `TelescopeFrame`, we don't need to change telescope frame for this, as the attributes needed for the transformation can be specified on the `SkyCoord` as seen here:

https://github.com/cta-observatory/ctapipe/blob/8a7a367dfe4a2f4e9bdf0e9bf774be29bf03b75f/ctapipe/visualization/mpl_camera.py#L434-L439

Comment (kosack) [reply ctapipe/image/extractor.py]: Remember there are in principle rotation corrections and maybe others, so you have to apply pointing corrections _before_ Hillas parameterization (doing it while going to Alt/Az wouldn't work), as you need to have correctly shifted/rotated parameters before you do shower geometry reconstruction.  But that can be solved when the ShowerProcessor is moved to TelescopeFrame later (right now e.g. HillasReconstructor supports both frames, but that's ugly).

File: ctapipe/image/tests/test_hillas.py [hunk 8]
```diff
@@ -3,26 +3,23 @@
...
 from ctapipe.instrument import CameraGeometry, SubarrayDescription
 
 
 def create_sample_image(
     psi="-30d",
-    x=0.2 * u.m,
-    y=0.3 * u.m,
-    width=0.05 * u.m,
-    length=0.15 * u.m,
+    x=0.2 * u.deg,
```
Comment (maxnoe) [inline ctapipe/image/tests/test_hillas.py:19]: Did you check that these images make sense when just changing meters to degrees?

File: ctapipe/image/toymodel.py [hunk 9]
```diff
@@ -80,8 +87,8 @@ def obtain_time_image(x, y, centroid_x, centroid_y, psi, time_gradient, time_int
 
     """
     longitudinal, _ = camera_to_shower_coordinates(x, y, centroid_x, centroid_y, psi)
-    longitudinal_m = longitudinal.to_value(u.m)
-    time_gradient_ns_m = time_gradient.to_value(u.ns / u.m)
+    longitudinal_m = longitudinal.to_value(u.deg)
```
Comment (maxnoe) [inline ctapipe/image/toymodel.py:90]: variable names still contain the old units

File: docs/changes/2134.bug.rst [hunk 10]
```diff
@@ -0,0 +1 @@
+Fix for Hillas lines in ``ArrayDisplay`` being wrong in the new ``EastingNorthingFrame``.
```
Comment (maxnoe) [inline docs/changes/2134.bug.rst:1]: Is that part of this PR?

File: docs/changes/2155.bug.rst [hunk 11]
```diff
@@ -0,0 +1 @@
+Replace usage of ``$HOME`` with ``Path.home()`` for cross-platform compatibility.
```
Comment (maxnoe) [inline docs/changes/2155.bug.rst:1]: Not part of this PR

File: docs/changes/2134.bug.rst (hunk 10 above)
Comment (StFroese) [reply docs/changes/2134.bug.rst:1]: I think something went wrong while rebasing. It's not part of the PR, I thinks it's an old one

File: ctapipe/image/tests/test_hillas.py (hunk 8 above)
Comment (StFroese) [reply ctapipe/image/tests/test_hillas.py:19]: I checked for the notebooks, I can recheck for all other files

**Comment by maxnoe** [comment]
superseded by #2396
